markdown/util.py

```diff
@@ -38,13 +38,283 @@
+        return _BlockLevelElements(self._list)
+
+    @deprecated("Using block level elements as a list is deprecated, use it as a set instead.")
+    def count(self, value: str, /) -> int:
+        # Count in list, for backwards compatibility.
+        # If used as a set, both counts will be the same (1).
+        return self._list.count(value)
+
+    def difference(self, *others: set[str]) -> set[str]:
+        # User expects a set back.
+        return self._set.difference(*others)
+
+    def difference_update(self, *others: set[str]) -> None:
+        # In-place difference should update both list and set.
+        self._set.difference_update(*others)
+        # Elements were only removed.
+        self._list[:] = [element for element in self._list if element in self._set]
+
+    def discard(self, element: str, /) -> None:
+        # In-place discard should update both list and set.
+        self._set.discard(element)
+        try:
+            self._list.remove(element)
+        except ValueError:
+            pass
+
+    @deprecated("Using block level elements as a list is deprecated, use it as a set instead.")
+    def extend(self, elements: list[str], /) -> None:
+        # In-place extension should update both list and set.
+        self._list.extend(elements)
+        self._set.update(elements)
+
+    @deprecated("Using block level elements as a list is deprecated, use it as a set instead.")
+    def index(self, value, start: int = 0, stop: int = sys.maxsize, /):
+        return self._list.index(value, start, stop)
+
+    @deprecated("Using block level elements as a list is deprecated, use it as a set instead.")
+    def insert(self, index: int, element: str, /) -> None:
+        # In-place insertion should update both list and set.
+        self._list.insert(index, element)
+        self._set.add(element)
+
+    def intersection(self, *others: set[str]) -> set[str]:
+        # User expects a set back.
+        return self._set.intersection(*others)
+
+    def intersection_update(self, *others: set[str]) -> None:
+        # In-place intersection should update both list and set.
+        self._set.intersection_update(*others)
+        # Elements were only removed.
+        self._list[:] = [element for element in self._list if element in self._set]
+
+    def isdisjoint(self, other: set[str], /) -> bool:
+        return self._set.isdisjoint(other)
+
+    def issubset(self, other: set[str], /) -> bool:
+        return self._set.issubset(other)
+
+    def issuperset(self, other: set[str], /) -> bool:
+        return self._set.issuperset(other)
+
+    def pop(self, index: int | None = None, /) -> str:
+        # In-place pop should update both list and set.
+        if index is None:
+            index = -1
+        else:
+            warnings.warn(
+                "Using block level elements as a list is deprecated, use it as a set instead.",
+                DeprecationWarning,
+            )
```

Comment:
We still use `warnings.warn` here because we don't want to deprecate the whole method, only the `index` parameter.